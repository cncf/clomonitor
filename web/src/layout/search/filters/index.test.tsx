import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Filters from './index';

const mockOnChange = jest.fn();

const defaultProps = {
  visibleTitle: true,
  activeFilters: {},
  onChange: mockOnChange,
  device: 'test',
};

describe('Filters', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<Filters {...defaultProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders Filters', () => {
      render(<Filters {...defaultProps} />);

      expect(screen.getByText('Filters')).toBeInTheDocument();

      expect(screen.getByText('Maturity level')).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'Graduated' })).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'Incubating' })).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'Sandbox' })).toBeInTheDocument();

      expect(screen.getByText('Rating')).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'A [75-100]' })).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'B [50-74]' })).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'C [25-49]' })).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'D [0-24]' })).toBeInTheDocument();

      expect(screen.getByText('Category')).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'App definition' })).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'Observability' })).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'Orchestration' })).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'Platform' })).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'Provisioning' })).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'Runtime' })).toBeInTheDocument();
      expect(screen.getByRole('checkbox', { name: 'Serverless' })).toBeInTheDocument();
    });

    it('renders Filters with selected options', () => {
      render(<Filters {...defaultProps} activeFilters={{ maturity: ['2'], rating: ['a', 'b'] }} />);

      expect(screen.getByRole('checkbox', { name: 'Sandbox' })).toBeChecked();
      expect(screen.getByRole('checkbox', { name: 'A [75-100]' })).toBeChecked();
      expect(screen.getByRole('checkbox', { name: 'B [50-74]' })).toBeChecked();
    });

    it('calls onChange to click filter', () => {
      render(<Filters {...defaultProps} />);

      const check = screen.getByRole('checkbox', { name: 'App definition' });

      expect(check).not.toBeChecked();

      userEvent.click(check);

      expect(mockOnChange).toHaveBeenCalledTimes(1);
      expect(mockOnChange).toHaveBeenCalledWith('category', '0', true);
    });
  });
});
