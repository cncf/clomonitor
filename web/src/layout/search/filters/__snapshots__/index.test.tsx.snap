// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-row align-items-center justify-content-between pb-2 mb-4 border-bottom"
  >
    <div
      class="h6 text-uppercase mb-0 lh-base text-primary fw-bold"
    >
      Filters
    </div>
  </div>
  <div
    class="fw-bold text-uppercase text-primary categoryTitle"
  >
    <small>
      Maturity level
    </small>
  </div>
  <div
    class="mt-2"
  >
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-maturity-0"
        name="maturity"
        tabindex="0"
        type="checkbox"
        value="0"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-maturity-0"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          />
          <span
            class="text-truncate"
          >
            Graduated
          </span>
        </div>
      </label>
    </div>
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-maturity-1"
        name="maturity"
        tabindex="0"
        type="checkbox"
        value="1"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-maturity-1"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          />
          <span
            class="text-truncate"
          >
            Incubating
          </span>
        </div>
      </label>
    </div>
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-maturity-2"
        name="maturity"
        tabindex="0"
        type="checkbox"
        value="2"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-maturity-2"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          />
          <span
            class="text-truncate"
          >
            Sandbox
          </span>
        </div>
      </label>
    </div>
  </div>
  <div
    class="fw-bold text-uppercase text-primary categoryTitle"
  >
    <small>
      Rating
    </small>
  </div>
  <div
    class="mt-2"
  >
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-rating-a"
        name="rating"
        tabindex="0"
        type="checkbox"
        value="a"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-rating-a"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          >
            <div
              class="me-2 border rounded-pill position-relative quality level1 undefined"
              data-testid="quality-dot"
            />
          </span>
          <span
            class="text-truncate"
          >
            A
          </span>
          <small
            class="ps-1"
          >
            [75-100]
          </small>
        </div>
      </label>
    </div>
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-rating-b"
        name="rating"
        tabindex="0"
        type="checkbox"
        value="b"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-rating-b"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          >
            <div
              class="me-2 border rounded-pill position-relative quality level2 undefined"
              data-testid="quality-dot"
            />
          </span>
          <span
            class="text-truncate"
          >
            B
          </span>
          <small
            class="ps-1"
          >
            [50-74]
          </small>
        </div>
      </label>
    </div>
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-rating-c"
        name="rating"
        tabindex="0"
        type="checkbox"
        value="c"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-rating-c"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          >
            <div
              class="me-2 border rounded-pill position-relative quality level3 undefined"
              data-testid="quality-dot"
            />
          </span>
          <span
            class="text-truncate"
          >
            C
          </span>
          <small
            class="ps-1"
          >
            [25-49]
          </small>
        </div>
      </label>
    </div>
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-rating-d"
        name="rating"
        tabindex="0"
        type="checkbox"
        value="d"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-rating-d"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          >
            <div
              class="me-2 border rounded-pill position-relative quality level4 undefined"
              data-testid="quality-dot"
            />
          </span>
          <span
            class="text-truncate"
          >
            D
          </span>
          <small
            class="ps-1"
          >
            [0-24]
          </small>
        </div>
      </label>
    </div>
  </div>
  <div
    class="fw-bold text-uppercase text-primary categoryTitle"
  >
    <small>
      Category
    </small>
  </div>
  <div
    class="mt-2"
  >
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-category-0"
        name="category"
        tabindex="0"
        type="checkbox"
        value="0"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-category-0"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          />
          <span
            class="text-truncate"
          >
            App definition
          </span>
        </div>
      </label>
    </div>
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-category-1"
        name="category"
        tabindex="0"
        type="checkbox"
        value="1"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-category-1"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          />
          <span
            class="text-truncate"
          >
            Observability
          </span>
        </div>
      </label>
    </div>
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-category-2"
        name="category"
        tabindex="0"
        type="checkbox"
        value="2"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-category-2"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          />
          <span
            class="text-truncate"
          >
            Orchestration
          </span>
        </div>
      </label>
    </div>
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-category-3"
        name="category"
        tabindex="0"
        type="checkbox"
        value="3"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-category-3"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          />
          <span
            class="text-truncate"
          >
            Platform
          </span>
        </div>
      </label>
    </div>
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-category-4"
        name="category"
        tabindex="0"
        type="checkbox"
        value="4"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-category-4"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          />
          <span
            class="text-truncate"
          >
            Provisioning
          </span>
        </div>
      </label>
    </div>
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-category-5"
        name="category"
        tabindex="0"
        type="checkbox"
        value="5"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-category-5"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          />
          <span
            class="text-truncate"
          >
            Runtime
          </span>
        </div>
      </label>
    </div>
    <div
      class="form-check me-sm-2 mb-2 undefined"
    >
      <input
        aria-checked="false"
        class="form-check-input rounded-0 checkbox"
        id="test-category-6"
        name="category"
        tabindex="0"
        type="checkbox"
        value="6"
      />
      <label
        class="form-check-label mw-100"
        data-testid="checkboxLabel"
        for="test-category-6"
      >
        <div
          class="d-flex align-items-baseline mw-100"
        >
          <span
            class="position-relative decorator"
          />
          <span
            class="text-truncate"
          >
            Serverless
          </span>
        </div>
      </label>
    </div>
  </div>
</DocumentFragment>
`;
